### ISSUE TITLE
NSA Kubernetes Hardening Guidance as a kb-resource

### ISSUE THREAD

**Issue by ioggstream** [d3fend/d3fend-ontology#32]
## Question

Can https://www.nsa.gov/Press-Room/News-Highlights/Article/Article/2716980/nsa-cisa-release-kubernetes-hardening-guidance/ be used as a reference for techniques?

**Comment by netfl0** [comment]
Yes, absolutely.

**Comment by ioggstream** [comment]
This guide details the following mitigations:
 Scan containers and Pods for vulnerabilities or misconfigurations. d3f:ApplicationHardening, d3f:ApplicationConfigurationHarndening `?SystemConfigurationHardening != d3f:SystemInitConfigAnalysis` 
 Run containers and Pods with the least privileges possible. d3f:ExecutionIsolation
 Use network separation to control the amount of damage a compromise can
cause. d3f:NetworkIsolation
 Use firewalls to limit unneeded network connectivity [d3f:NetworkIsolation] and use encryption to
protect confidentiality. [d3f:MessageEncryption]
 Use strong authentication and authorization to limit user and administrator
access as well as to limit the attack surface. d3f:CredentialHardening
 Capture and monitor audit logs so that administrators can be alerted to potential
malicious activity. d3f:Detect
 Periodically review all Kubernetes settings and use vulnerability scans to ensure
risks are appropriately accounted for and security patches are applied. d3f:Harden or d3f:Detect ?

**Comment by netfl0** [comment]
I was having unicode issue with that message for some reason, repost with comments:

This guide details the following mitigations:
- Scan containers and Pods for vulnerabilities or misconfigurations. d3f:ApplicationHardening, d3f:ApplicationConfigurationHarndening ?SystemConfigurationHardening != d3f:SystemInitConfigAnalysis
> Very broad directive, I agree with classifications
- Run containers and Pods with the least privileges possible. d3f:ExecutionIsolation
> There are multiple ways to limit application privileges. We might be missing one wrt to user account running the process. Potential new technique here.
- Use network separation to control the amount of damage a compromise can
cause. d3f:NetworkIsolation
> [D3-BDI (Broadcast Domain Isolation)](https://d3fend.mitre.org/technique/d3f:BroadcastDomainIsolation) and forthcoming D3-NTF
- Use firewalls to limit unneeded network connectivity [d3f:NetworkIsolation] and use encryption to
protect confidentiality. [d3f:MessageEncryption]
> I'd go with [D3-ET](https://d3fend.mitre.org/technique/d3f:EncryptedTunnels) over D3-ME, and also forthcoming D3-NTF.
- Use strong authentication and authorization to limit user and administrator
access as well as to limit the attack surface. d3f:CredentialHardening
- Capture and monitor audit logs so that administrators can be alerted to potential
malicious activity. d3f:Detect
> Not specific enough guidance to model with D3FEND. E.g. monitor for what specifically.
- Periodically review all Kubernetes settings and use vulnerability scans to ensure
risks are appropriately accounted for and security patches are applied. d3f:Harden or d3f:Detect ?
> [D3-ACH](https://d3fend.mitre.org/technique/d3f:ApplicationConfigurationHardening), [D3-SU](https://d3fend.mitre.org/technique/d3f:SoftwareUpdate)

**Comment by netfl0** [comment]
> 



> ### SystemConfigurationHardening is missing
> 
> > > Scan containers and Pods for vulnerabilities or misconfigurations.
> > > d3f:ApplicationHardening, d3f:ApplicationConfigurationHarndening
> > > Very broad directive, I agree with classifications
> 
> Currently we don't have a preventive measure (e.g. SystemConfigurationHardening) but only a detective misure: d3f:SystemInitConfigAnalysis
> 
> should we add SystemConfigurationHardening ?

Yes, though we recently added this to address https://github.com/d3fend/d3fend-ontology/issues/19
Platform and system are synonyms, we chose platform as primary since system can more easily be considered be multi-host.

> ### use encryption to protect confidentiality
> 
> This can be done at different levels:
> 
>     * network, but   d3f:EncryptedTunnel definition seems to just cover IPsec tunnel, not IPSec transport+Gre,  not TLS, not conflated transport+encryption protocols
> 
>     * application (an encrypted file/content)

Yes agree, given the reference to firewalls I was thinking primarily VPN. D3-ET should cover TLS. We may want to add some sub-classes of encrypted tunnels to enable more specificity.

**Comment by netfl0** [comment]
@ioggstream are there any action items you want to see here?

**Comment by netfl0** [comment]
Considering this covered by latest release.
